### DISCUSSION TITLE
How can I achieve that the chat history of previous pages is not displayed on subsequent pages?

### DISCUSSION THREAD

**Discussion by jbengelsdorf** [ctreffe/alfred#109] [Alfred Q&A] [answered]
The alfred3-plugin [alfred3-interact](https://github.com/jobrachem/alfred3-interact) provides a chat window that can be used for real-time chat as well as for a kind of message board. When I add the [Chat](https://jobrachem.github.io/alfred3-interact/build/html/generated/alfred3_interact.element.Chat.html?highlight=chat#alfred3_interact.element.Chat) to multiple pages, the chat history of the previous pages is currently displayed on the subsequent pages. How can I clear the chat history of a group so that the chat history of the previous pages is not displayed on the subsequent pages?

**Comment by jobrachem** [comment, marked as answer]
Starting from alfred3-interact v0.1.9, the `Chat` element and the `Group.chat` short-hand method for spawning a group chat offer the argument `room` as the most convenient way to achieve this result. You can define any string as the chat room, so the page name is one good possibility to achieve your desired result.

Below is a minimal example experiment:

```python
import alfred3 as al
import alfred3_interact as ali

exp = al.Experiment()

@exp.setup
def setup(exp):
    mm = ali.MatchMaker("a", "b", exp=exp)
    exp.plugins.mm = mm


@exp.member
class Match(ali.MatchingPage):

    def wait_for(self):
        group = self.exp.plugins.mm.match_groupwise()
        self.exp.plugins.group = group
        return True


@exp.member
class ChatPage1(al.Page):
    title = "Chat Page 1"

    def on_first_show(self):
        group = self.exp.plugins.group
        self += group.chat(room=self.name)


@exp.member
class ChatPage2(al.Page):
    title = "Chat Page 2"

    def on_first_show(self):
        group = self.exp.plugins.group
        self += group.chat(room=self.name)
```
